File: 0965-unique-email-addresses/0965-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:

        unique_emails = set()

        for email in emails:
            unique_emails.add(self.emailParser(email))
        
        return len(unique_emails)
    
    def emailParser(self, email):

        local, domain = email.split("@")
        parsed = []

        for char in local:
            if char == "+":
                break
            if char == ".":
                continue
            parsed.append(char)
        
        new_email = "".join(parsed) + "@" + domain
        return new_email
    

        

File: 0965-unique-email-addresses/0965-unique-email-addresses.cs
public class Solution {
    public int NumUniqueEmails(string[] emails) {
        var seen = new HashSet<string>();
        int cnt = 0;
        foreach(var email in emails) {
            var s = processEmail(email);
            if (!seen.Contains(s)) cnt++;
            seen.Add(s);
        }
        return cnt;
    }

    private string processEmail(string email) {
        string[] s = email.Split("@");
        var sb = new StringBuilder("");
        foreach (char c in s[0]) {
            if (c == '.') continue;
            if (c == '+') break;
            sb.Append(c);
        }
        return sb.Append("@" + s[1]).ToString();
    }
}